### PR TITLE
Set LeaderElectionNamespace to PodNamespace

### LINKED ISSUE #5404: Running leader election on KEDA running outside of cluster mode
### Report

Similar to what we experienced running Out Of Cluster mode in Karpenter [here](https://github.com/aws/karpenter-provider-aws/issues/4216), when trying to run leader election in OutOfCluster mode (i.e when we pass a kubeconfig file as a launch arg), the KEDA operator crashes on launch with
```
unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace
```

I believe for KEDA since we have `kedautil.GetPodNamespace()` already, we can add
```
LeaderElectionNamespace: kedautil.GetPodNamespace(),
```
to the controller manager initializer .

### Expected Behavior

When running in OutOfCluster mode with a passed in Kubeconfig, Leader Election should work.

### Actual Behavior

KEDA crashes on launch with 
```
unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace
```

### Steps to Reproduce the Problem

1. Run KEDA-Operator in OutOfCluster mode by passing in kubeconfig file as launch arg
2. Enable Leader Election
3.

### Logs from KEDA operator

```
unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace
```

### KEDA Version

None

### Kubernetes Version

1.25

### Platform

Amazon Web Services

### Scaler Details

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by aliaqel-stripe**
Sets the LeaderElectionNamespace on the controller runtime in keda-operator to use the PodNamespace so that leader election works in OutOfCluster mode. This is based on a[ similar change we had to make in karpenter](https://github.com/kubernetes-sigs/karpenter/pull/424)

Couldn't find any general tests that test the operator but since this uses an existing utility and a parameter passed to the controller runtime, I don't think it's changing behavior enough to KEDA that it warrants a new test, but let me know and I'll happily write a new test!

I did validate manually in my own cluster and this worked.

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5404

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Here is an overview of all new **experimental** features:
 
 ### Fixes
 
+- **General**: Set LeaderElectionNamespace to PodNamespace ([#5404](https://github.com/kedacore/keda/issues/5404))
```
Comment (zroubalik) [inline CHANGELOG.md:69]:
```suggestion
- **General**: Set `LeaderElectionNamespace` to PodNamespace so leader election works in OutOfCluster mode ([#5404](https://github.com/kedacore/keda/issues/5404))
```

We are shipping 2.13, we can include this, just move it down below to 2.13, Improvements section

**Review by zroubalik — COMMENTED** [review]
Looking good, just a nit in the changelog

File: CHANGELOG.md (hunk 1 above)
Comment (aliaqel-stripe) [reply CHANGELOG.md:69]: Thanks, wasn't sure as 2.13 is coming out tomorrow :)

Comment (aliaqel-stripe) [reply CHANGELOG.md]: updated in my latest commit

**Review by zroubalik — APPROVED** [review]

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**